task/thermal-api/src/lib.rs

```diff
@@ -26,6 +26,7 @@ pub enum ThermalError {
     InvalidWatchdogTime = 7,
     InvalidParameter = 8,
     InvalidIndex = 9,
+    FanControllerUninitialized = 10,
```

Comment:
**question:** Just curious but is there a particular reason to define the values after 1? Is it to keep enum values backwards compatible in the face of possible removals?

Reply:
I believe this is for use with C-like error types, where we use 0 to represent "no error".

Reply:
Ah, I meant why keep defining `=2` etc instead of letting Rust lay those out in numeric order after the `=1`.

Reply:
Ah, I see. Yeah, we *could* just use the auto-increment numbering and it would be equivalent, I guess it's just a matter of taste. Personally, I kind of like the explicitness, you can look at the code and see which variant corresponds to a given number more easily without having to count variants yourself. But, I don't think it matters a whole lot either way.

Reply:
Ah okay, makes sense, especially if number-to-code reverse lookup is done at any regularity.

Thank you for humouring my curiosity.

Reply:
I don't think it's something we expect to do super often, but it certainly could happen while e.g. debugging a crash dump. Although, I wasn't the one who originally decided to use explicit numbers here, so, I'm mostly just following the existing code for consistency. :)